src/app/feature/calendar/cards/calendar-card-factory.directive.ts

```diff
@@ -54,15 +55,14 @@ export const christmasCardsFactoryMap: Record<ChristmasCardType, Type<ChristmasC
 
 @Directive({
   selector: '[xmasCalendarCardFactory]',
-  standalone: true,
 })
 export class CalendarCardFactoryDirective implements OnInit {
-  @Input({required: true}) type!: ChristmasCardType;
+  readonly type: InputSignal<ChristmasCardType> = input.required<ChristmasCardType>();
```

Comment:
I think you don't need type twice

```suggestion
  readonly type: InputSignal<ChristmasCardType> = input.required();
```

Reply:
That my personal preference, I can remove the generic type or let infer the type if you really want this change?

Reply:
Sure I will change this ^^